crud.c

```diff
@@ -80,10 +80,14 @@ void set_rects_from_selection(XRectangle* rects, Selection dimensions) {
 
 void switch_cursor(Cursor* cursor) {
   int mask = PointerMotionMask | ButtonPressMask | ButtonReleaseMask;
-
-  XGrabPointer(display, root, True,
-               mask, GrabModeAsync, GrabModeAsync,
-               None, *cursor, CurrentTime);
+  int error;
+
+  do {
+    error = XGrabPointer(display, root, True,
+                 mask, GrabModeAsync, GrabModeAsync,
+                 None, *cursor, CurrentTime);
+    sleep(0);
```

Comment:
i haven't had time to actually try this out yet, but is there a justification for sleeping here?